### PR TITLE
👷(deployment) add github workflow for deployment

### PR DESCRIPTION
**Pull request by Kitero**
## Purpose

Add workflow to deploy automatically on the branch pages-deployment when main is updated

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ aliases:
         - v2-front-dependencies-{{ checksum "~/fun/yarn.lock" }}
         - v2-front-dependencies-
 
-# Define the jobs we want to run for this project
+# Define the jobs we want to run for this project
```
Comment (jbpenrath) [inline .circleci/config.yml:15]: Rollback this change
```suggestion
# Define the jobs we want to run for this project
```

File: examples/playground/index.html [hunk 2]
```diff
@@ -7,6 +7,6 @@
   </head>
   <body>
     <div id="root"></div>
-    <script type="module" src="/src/main.tsx"></script>
+    <script type="module" src="./src/main.tsx"></script>
```
Comment (jbpenrath) [inline examples/playground/index.html:10]: Debug residue or really required ?

File: lib/vite.config.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default defineConfig({
       entry: resolve(__dirname, './src/index.ts'),
       fileName: () => 'index.js',
       formats: ['es'],
-      name: 'index',
+      name: 'verna',
```
Comment (jbpenrath) [inline lib/vite.config.ts:13]: Debug residue I think

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
   "private": true,
   "scripts": {
     "build": "yarn workspace @openfun/verna build",
+    "build:playground": "yarn workspace verna-examples-playground build",
```
Comment (jbpenrath) [inline package.json:23]: Like for `dev:example`, I'll make this command more generic
```suggestion
    "build:example": "yarn workspace ${VERNA_EXAMPLE:-verna-examples-playground} build",
```

File: examples/playground/vite.config.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import { resolve } from 'path';
-import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react';
+import { defineConfig } from 'vite';
 
 // https://vitejs.dev/config/
 export default defineConfig({
+  base: '/verna/',
```
Comment (jbpenrath) [inline examples/playground/vite.config.ts:7]: No impact for dev purpose ?

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Added
 
+- Add auto deployment of main in circle-ci
```
Comment (jbpenrath) [inline CHANGELOG.md:13]: As this work does not impract production code, add an entry into the CHANGELOG is not recommended.

File: .circleci/config.yml [hunk 7]
```diff
@@ -155,6 +155,60 @@ jobs:
...
+          name: Install dependencies and build Verna for production
+          command: |
+            yarn install --frozen-lockfile
+            yarn build
+      - save_cache:
+          paths:
+            - ./lib/node_modules
+            - ./examples/playground/node_modules
+            - ./node_modules
+          key: v2-front-dependencies-{{ checksum "yarn.lock" }}
```
Comment (jbpenrath) [inline .circleci/config.yml:183]: As `deployment` job is executed after `test` job, the lib has been already built and all dependencies have been installed. Within the job you restore those outcomes through the `*restore_node_modules`. So it is useless to install and build library again.

File: examples/playground/index.html (hunk 2 above)
Comment (Kitero) [reply examples/playground/index.html:10]: debug residue :p

File: examples/playground/vite.config.ts (hunk 5 above)
Comment (Kitero) [reply examples/playground/vite.config.ts:7]: Nop, all good

File: .circleci/config.yml (hunk 7 above)
Comment (Kitero) [reply .circleci/config.yml:183]: Well seen ! I deleted everything you quoted

File: .circleci/config.yml [hunk 8]
```diff
@@ -155,6 +155,49 @@ jobs:
...
+          fingerprints:
+            - "cd:48:76:dd:42:15:64:18:7c:1a:31:2b:d6:ab:60:57"
+      - *checkout_fun
+      - *restore_node_modules
+      # If the yarn.lock file is not up-to-date with the package.json file,
+      # using the --frozen-lockfile should fail.
+      - run:
+          name: build Playground for production
+          command: |
+            yarn build:playground
```
Comment (jbpenrath) [inline .circleci/config.yml:176]:
```suggestion
            VERNA_EXAMPLE=verna-examples-playground yarn build:example
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -155,6 +155,49 @@ jobs:
...
+          password: $DOCKER_PASS
+    working_directory: ~/fun
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "cd:48:76:dd:42:15:64:18:7c:1a:31:2b:d6:ab:60:57"
+      - *checkout_fun
+      - *restore_node_modules
+      # If the yarn.lock file is not up-to-date with the package.json file,
+      # using the --frozen-lockfile should fail.
```
Comment (jbpenrath) [inline .circleci/config.yml:172]: This comment is no more relevant
```suggestion
```

**Comment by jbpenrath** [comment]
Add the link of the demo at the top of the README : )

**Comment by Kitero** [comment]
Add two folders, one for each example

File: README.md [hunk 10]
```diff
@@ -11,7 +16,7 @@ library, which makes it fully extensible to fit your needs.
 
 
 ## Quick preview
-
+
```
Comment (jbpenrath) [inline README.md:19]:
```suggestion
 
```

File: README.md [hunk 11]
```diff
@@ -3,6 +3,11 @@
 
 An extensible form builder based on [React JSON Schema Form](https://github.com/rjsf-team/react-jsonschema-form).
 
+## Demos
+
+Basic form playground: https://openfun.github.io/verna/basic
+Poll playground: https://openfun.github.io/verna/poll
```
Comment (jbpenrath) [inline README.md:9]: I think it's better to add the second link into the grommet PR

Comment (jbpenrath) [inline README.md:9]:
```suggestion
## Demos

To demonstrate verna capabilities, we developed several use case examples.

- Basic form playground:[👀 Live demo](https://openfun.github.io/verna/basic), [🧑‍💻 Source code](https://github.com/openfun/verna/tree/main/examples/playground)
- Poll playground: [👀 Live demo](https://openfun.github.io/verna/basic), [🧑‍💻 Source code](https://github.com/openfun/verna/tree/main/examples/playground)
```

File: .circleci/config.yml [hunk 12]
```diff
@@ -155,6 +155,47 @@ jobs:
...
+            cp -R .git ~/deployment/
+            cd ~/deployment
+            git fetch
+            git checkout pages-deployment
+      - run:
+          name: Clean the deployment directory & copy the build results
+          command: |
+            rm -rf ~/deployment/*
+            mkdir ~/deployment/basic
+            cp -R examples/playground/dist/* ~/deployment/basic/
```
Comment (jbpenrath) [inline .circleci/config.yml:188]: To ease things, I think its better to use the same directory name as the example
```suggestion
            cp -R examples/playground/dist/* ~/deployment/playground/
```

Comment (jbpenrath) [inline .circleci/config.yml:188]: No way to build and copy all examples ?

File: README.md (hunk 11 above)
Comment (Kitero) [reply README.md:9]: I added only the Basic form then for now, i'll had the other one on the pr of grommet

File: .circleci/config.yml [hunk 13]
```diff
@@ -154,6 +154,58 @@ jobs:
...
+            - "cd:48:76:dd:42:15:64:18:7c:1a:31:2b:d6:ab:60:57"
+      - *checkout_fun
+      - *restore_node_modules
+      - run:
+          name: Build front-end application for production
+          command: yarn build
+      - run:
+          name: build examples for production
+          command: |
+            for workspace in $(yarn workspaces info | grep -oP "verna-examples.*(?=\":)")
```
Comment (jbpenrath) [inline .circleci/config.yml:176]: 🏅

File: examples/playground/vite.config.ts [hunk 14]
```diff
@@ -1,9 +1,10 @@
 import { resolve } from 'path';
-import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react';
+import { defineConfig } from 'vite';
 
 // https://vitejs.dev/config/
 export default defineConfig({
+  base: '/verna/playground/',
```
Comment (jbpenrath) [inline examples/playground/vite.config.ts:7]:
```suggestion
  // Defined to match the path to access to this demo on github pages. (https://openfun.github.io/verna/playground/)
  base: '/verna/playground/',
```

**Review by jbpenrath — APPROVED** [review]
Last comments, rebase then merge it 💪

**Pull request merged**